aderyn_core/src/detect/low/missing_inheritance.rs

```diff
@@ -0,0 +1,192 @@
+                    continue;
+                }
+            }
+            let inheritances = inheritance_map.entry(*contract_id).or_default();
+            for (potentially_missing_inheritance, missing_function_selectors) in
+                &contract_function_selectors
+            {
+                // Check that it's not empty
+                if missing_function_selectors.is_empty() {
+                    continue;
+                }
+
+                // Check that it's not the same contract
+                if potentially_missing_inheritance == contract_id {
+                    continue;
+                }
+
+                // Check that it's not already inherited
+                if inheritances.contains(potentially_missing_inheritance) {
+                    continue;
+                }
+
+                if let Some(ASTNode::ContractDefinition(c)) =
+                    context.nodes.get(potentially_missing_inheritance)
+                {
+                    if c.kind == ContractKind::Interface || c.is_abstract.map_or(false, identity) {
+                        // Check that the contract is compatible with the missing inheritance
+                        if missing_function_selectors
+                            .iter()
+                            .all(|s| contract_selectors.contains(s))
+                        {
+                            results
+                                .entry(*contract_id)
+                                .or_default()
+                                .insert(c.name.clone());
+                        }
+                    }
+                }
+            }
+        }
+
+        for (contract, missing_inheritances) in results {
+            if let Some(ASTNode::ContractDefinition(c)) = context.nodes.get(&contract) {
+                let missing_inheritances_vector =
+                    missing_inheritances.iter().cloned().collect::<Vec<_>>();
+                let missing_inheritaces_string = missing_inheritances_vector.join(", ");
+
+                let hint = format!(
+                    "Consider implementing the most suitable of following inheritances : {}",
+                    missing_inheritaces_string
+                );
+
+                capture!(self, context, c, hint);
+            }
+        }
+
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::Low
+    }
+
+    fn hints(&self) -> BTreeMap<(String, usize, String), String> {
+        self.hints.clone()
+    }
+
+    fn title(&self) -> String {
+        String::from("Potentially missing inheritance for contract.")
+    }
+
+    fn description(&self) -> String {
+        String::from("There is an interface / abstract contract that is potentially missing (not included in) the inheritance of this contract. If that's not the case, consider using the same interface instead of defining multiple identical interfaces.")
+    }
```

Comment:
Do you agree with this @alexroan  (the last part) ?

Reply:
If not, we can adjust the logic to report missing inheritance only if it defines unique functions such that no other interface that is implemented, defines them.

Reply:
IDEA - I Will restrict this detector to run on contracts that have don't inherit from anywhere. This way we are respecting the developer's choice